projects/core/src/model/misc.model.ts

```diff
@@ -128,4 +128,10 @@ export interface BaseSite {
   baseStore?: BaseStore;
   requiresAuthentication?: boolean;
   isolated?: boolean;
+  captchaConfig?: CaptchaConfig;
+}
+
+export interface CaptchaConfig {
```

Comment:
I had a look at this and the instructions. It doesn't seem to be indicated how the user can turn on Captcha with this part of the config (ie. set `enabled` to true). I also wonder why the users need to override the CaptchaModule's default config. Wouldn't it be easier for them to enable and provide `CaptchaConfig` in their own spartacus config module like we do in `SpartacusB2cConfigurationModule`?

Reply:
As you known, Whether captcha is enabled or not is defined in backend and read it via OCC API. We need a configuration to present what's we get from backend.  The is Enabled is a mandatory filed of it. PublicKey is set as optional as extra parameters.

Check the end points"/getBaseSites"  https://api.cg79x9wuu9-eccommerc1-d2-public.model-t.myhybris.cloud/occ/v2/swagger-ui/index.html#/Base%20Sites/getBaseSites.  you can see our OOTB baseSIteInfo contains "enabled" and "publicKey".  

 If customer wanna to integrate their backend, they may add other things more than publicKey(like URL, domain, ...) But we just define it as what OCC API provided.
<img width="741" alt="image" src="https://github.com/SAP/spartacus/assets/38693320/cc6fee84-5ddb-4824-95a6-4f61dddb1284">

Reply:
For your question, how to enable, it mentioned in help doc as "Requirements" https://help.sap.com/docs/INTERNAL_COMPOSABLE_STOREFRONT/eaef8c61b6d9477daf75bff9ac1b7eb4/4605061113284c2c92cc558600140aaa.html?locale=en-US&state=DRAFT#loio4605061113284c2c92cc558600140aaa

You have enabled CAPTCHA for your site. For more information, see [CAPTCHA OCC APIs](https://help.sap.com/docs/INTERNAL_COMPOSABLE_STOREFRONT/eaef8c61b6d9477daf75bff9ac1b7eb4/8a42681cd28549b6a81d6faffc7d6b92.html?locale=en-US&state=DRAFT).

Reply:
The full picture of the CAPTCHA should looks as below

1) The feature toogle is in backend;
2) If CAPTCHA enabled, in SPA side, we'll render CAPTCHA component, and  generate token as designed;
3) SPA call APIs with adding CAPTCHA token in the header;
4) The backend side, need add annotation if one API need check the token; then the annotated API will check CAPTCHA token as part of authentication before business process.

In code in SPA, we only can see 2 and 3, Which make it little complex. But I'm not sure where we can make a overall picture in help doc as now we just put a linkage from SPA doc to Commerce(not sure that's enough)

Reply:
So maybe there is some confusion, sorry. There was two points here I would like to address:

- I understand that it needs to be enabled in the backend and is a baseSite endpoint. But it isn't clear how to enable this in Spartacus from the documentation. Maybe was meant to be in [this link](https://help.sap.com/docs/INTERNAL_COMPOSABLE_STOREFRONT/eaef8c61b6d9477daf75bff9ac1b7eb4/8a42681cd28549b6a81d6faffc7d6b92.html?locale=en-US&state=DRAFT)?

- I also think its better to tell customers to override the default captcha config from their own configuration module. I think it is unnecessary to override the CaptchaModule itself. I think this is our standard practice. Is there a reason to do it the way in the doc? Maybe it isn't meant to be customer facing?

Reply:
The doc update will be handled by UA in coming workdays.

Let's check it again once ready.